pkg/commands/access/actions/use.go

```diff
@@ -2,94 +2,109 @@ package actions
 
 import (
 	"bytes"
+	"context"
 	"encoding/json"
 	"errors"
 	"fmt"
 	"os/exec"
 	"runtime"
 	"strings"
 
+	"github.com/apono-io/apono-cli/pkg/utils"
+
 	"github.com/apono-io/apono-cli/pkg/clientapi"
+	"github.com/apono-io/apono-cli/pkg/interactive"
 
 	"github.com/spf13/cobra"
 
 	"github.com/apono-io/apono-cli/pkg/aponoapi"
 )
 
 const (
-	outputFlagName = "output"
-	runFlagName    = "run"
+	outputFlagName        = "output"
+	runFlagName           = "run"
+	noInteractiveFlagName = "no-interactive"
 
 	cliOutputFormat          = "cli"
 	linkOutputFormat         = "link"
 	instructionsOutputFormat = "instructions"
 	jsonOutputFormat         = "json"
 )
 
+type accessUseCommandFlags struct {
+	outputFormat               string
+	shouldExecuteAccessCommand bool
+	dontRunInteractive         bool
+}
+
 func AccessDetails() *cobra.Command {
-	var outputFormat string
-	var shouldExecuteAccessCommand bool
+	cmdFlags := &accessUseCommandFlags{}
 
 	cmd := &cobra.Command{
-		Use:   "use <id>",
+		Use:   "use <session_id>",
 		Short: "Get access session details",
-		Args:  cobra.MinimumNArgs(1), // This will enforce that exactly 1 argument must be provided
 		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 && cmdFlags.dontRunInteractive {
+				return fmt.Errorf("session id is required when using --%s flag", noInteractiveFlagName)
+			}
+
 			client, err := aponoapi.GetClient(cmd.Context())
 			if err != nil {
 				return err
 			}
 
-			sessionID := args[0]
-
-			session, _, err := client.ClientAPI.AccessSessionsAPI.GetAccessSession(cmd.Context(), sessionID).Execute()
-			if err != nil {
-				return fmt.Errorf("access session with id %s not found", sessionID)
+			var session *clientapi.AccessSessionClientModel
+			if len(args) == 0 && !cmdFlags.dontRunInteractive {
+				session, err = interactive.RunSessionsSelector(cmd.Context(), client)
+				if err != nil {
+					return err
+				}
+			} else {
+				sessionID := args[0]
+				session, _, err = client.ClientAPI.AccessSessionsAPI.GetAccessSession(cmd.Context(), sessionID).Execute()
+				if err != nil {
+					return fmt.Errorf("access session with id %s not found", sessionID)
+				}
 			}
 
-			if shouldExecuteAccessCommand {
```

Comment:
this makes the `apono access use <ID> --run` command to not work (opens interactive mode)

Reply:
fixed it, now it work